launchable/test_runners/maven.py

```diff
@@ -1,22 +1,47 @@
 import os
+from typing import List
+from xmlrpc.client import Boolean
 import click
 from . import launchable
 
 
-@click.argument('source_roots', required=True, nargs=-1)
+@click.option('--from-file', 'from_files', required=False, multiple=True, type=click.Path(exists=True))
+@click.argument('source_roots', required=False, nargs=-1)
 @launchable.subset
-def subset(client, source_roots):
+def subset(client, source_roots, from_files):
+
+    def is_file(f: str) -> Boolean:
+        return (f.endswith('.java') or f.endswith(".scala") or f.endswith(".kt"))
+
+    def file2class_test_path(f: str) -> List:
+        f = f[:f.rindex('.')]   # remove extension
```

Comment:
[nits]
`f = os.path.splitext(f)[0]`
`os.path.splitext` will split filename and extension, returning tuple.

```python
f = "aa/bb.txt"
f = os.path.splitext(f)
print(f)
# ('aa/bb', '.txt')

Reply:
Thanks, TIL: https://docs.python.org/3/library/os.path.html#os.path.splitext

Reply:
https://github.com/launchableinc/cli/pull/410/commits/521fdbd2857538514f50382f27efbfebefa20403